### PR TITLE
Upgrade to Blazor 3.2.0-preview2

### PR DESCRIPTION
**Pull request by jspuij**
- Upgraded to the latest Blazor.
- Fixed a bug in the latest webassembly where the gitter base Uri would not be set
- Converted to npm and gulp for sass processing.

### CODE REVIEW

File: src/Blazor.Gitter.Core/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    sourcemaps = require("gulp-sourcemaps"),
+    sass = require("gulp-sass"),
+    es = require('event-stream');
+
+gulp.task("sass", async function () {
+    var client = gulp.src('content/css/blazor.gitter.scss')
+        .pipe(sourcemaps.init())
+        .pipe(sass())
+        .pipe(sourcemaps.write('.'))
+        .pipe(gulp.dest('../Blazor.Gitter.Client/wwwroot/css'));
```
Comment (SQL-MisterMagoo) [inline src/Blazor.Gitter.Core/gulpfile.js:15]: Is it possible to keep the original compiled css name `blazored.gitter.css` ?

File: src/Blazor.Gitter.Core/gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    var server = gulp.src('content/css/blazor.gitter.scss')
+        .pipe(sourcemaps.init())
+        .pipe(sass())
+        .pipe(sourcemaps.write('.'))
+        .pipe(gulp.dest('../Blazor.Gitter.Server/wwwroot/css'));
```
Comment (SQL-MisterMagoo) [inline src/Blazor.Gitter.Core/gulpfile.js:23]: Is it possible to keep the original compiled css name blazored.gitter.css ?

File: src/Blazor.Gitter.Core/gulpfile.js (hunk 1 above)
Comment (jspuij) [reply src/Blazor.Gitter.Core/gulpfile.js:15]: Yeah, I can probably rename and minify it. Is it OK if I do this tomorrow? It's almost bedtime in my timezone ;-)

**Review by SQL-MisterMagoo — CHANGES_REQUESTED** [review]
Could you just see if we can get the original naming of the generated css file back to the original `blazored` naming?

I've noticed that the Webassembly version is not working correctly (not fetching older messages) and the css seems wrong on the server - BUT that could be how it was before the upgrade as I can see nothing has changed to cause that.

Once this is merged I'll take a look at why it's not right - I have another branch that I sometimes use to test things out and it may be I need to cherry pick some commits from there...

To be clear - this is not a problem with this PR - just letting you know so you don't waste time on it yourself.

**Review by SQL-MisterMagoo — CHANGES_REQUESTED** [review]
Just had a thought as well - the `azure-pipelines.yml` file will probably need to be updated to be able to build this- it's currently set to use `3.0.100-rc1-014190` but I think we will need `3.0.102` for this?

**Comment by chucker** [comment]
> Just had a thought as well - the `azure-pipelines.yml` file will probably need to be updated to be able to build this- it's currently set to use `3.0.100-rc1-014190` but I think we will need `3.0.102` for this?

You mean `3.1.102`. Also, we can't use VS 2017 any more, I'm assuming.

See #71. I'm not experienced at this, though.

**Comment by chrissainty** [comment]
> You mean 3.1.102. Also, we can't use VS 2017 any more, I'm assuming.

We need to upgrade the .NET Core installer task as well, the current one is being depreciated. The other repos in Blazored have had their yaml files updated, so setting can be copied from there. 

Just added a comment to #71

**Comment by SQL-MisterMagoo** [comment]
> You mean `3.1.102`. Also, we can't use VS 2017 any more, I'm assuming.
> 
> See #71. I'm not experienced at this, though.
Yep - you're right I got the version wrong, thanks!

**Comment by jspuij** [comment]
So I did the requested changes and accidentally also incorporated #71 as I missed that pull request. You should be able to pull this all at once.

**Review by SQL-MisterMagoo — APPROVED** [review]

**Comment by SQL-MisterMagoo** [comment]
@jspuij Sorry I just realised one more thing - 
Are there any new dependencies that would be required now that webcompiler has been replaced? Could you possibly update the README.md with details of those if there are any? 
then we should be good to go.

**Comment by SQL-MisterMagoo** [comment]
Thanks so much for doing this work!

**Pull request merged**